src/ILCompiler/src/Program.cs

```diff
@@ -181,7 +181,8 @@ private int Run(string[] args)
             typeSystemContext.InputFilePaths = _inputFilePaths;
             typeSystemContext.ReferenceFilePaths = _referenceFilePaths;
 
-            typeSystemContext.SetSystemModule(typeSystemContext.GetModuleForSimpleName(_systemModuleName));
+            typeSystemContext.SetSystemModule(typeSystemContext.GetModuleForSimpleName(_systemModuleName));           
+            typeSystemContext.SupportLazyCctors = typeSystemContext.SystemModule.GetType("System.Runtime.CompilerServices", "ClassConstructorRunner", false) != null;
```

Comment:
It does not look right to set this here. It should be done in the CompilerTypeSystemContext or similar.